ext/sockets/multicast.c

```diff
@@ -159,8 +159,16 @@ mcast_req_fun: ;
 			php_sockaddr_storage	group = {0};
 			socklen_t				glen;
 
-			convert_to_array(arg4);
-			opt_ht = Z_ARRVAL_P(arg4);
+			if (Z_TYPE_P(arg4) != IS_ARRAY) {
+				if (UNEXPECTED(Z_TYPE_P(arg4) != IS_OBJECT)) {
+					zend_type_error("must be of type array when argument #3 ($option) is MCAST_LEAVE_GROUP, %s given", zend_zval_value_name(arg4));
+					return FAILURE;
+				} else {
+					opt_ht = Z_OBJPROP_P(arg4);
+				}
+			} else {
```

Comment:
Maybe add using objects in these cases to the 8.5 deprecation RFC? https://wiki.php.net/rfc/deprecations_php_8_5